docs/operators/deduper.md

```diff
@@ -0,0 +1,523 @@
+Deduper - Operator Documentation
+=============
+
+Introduction
+=========================
+
+About this document
+--------------------------------
+
+This document is intended as a guide for understanding and using
+the Dedup operator.
+
+Terminology
+------------------------
+
+We will refer to this operator as the Deduper or Dedup operator
+interchangeably.
+
+Overview
+=====================
+
+Dedup - “What” in a Nutshell
+-----------------------------------------
+
+Dedup is actually short for Deduplication. Duplicates are omnipresent and
+can be found in almost any kind of data. Most of the times it is
+essential to discard, or at the very least separate out the data into
+unique and duplicate components. The entire purpose of this
+operator is to de-duplicate data. In other words, when data passes
+through this operator, it will be segregated into two different data
+sets, one of which contains all unique tuples, and the other which are
+occurring more than once in the original data set.
+
+
+
+![](images/deduper/image00.png)
+
+
+Dedup - “How” in a Nutshell
+----------------------------------------
+
+In order to quickly decide whether an incoming tuple is duplicate
+or unique, it has to store each incoming tuple (or a signature, like key
+for example) to be used for comparison later. A plain in-memory storage
+may work for small datasets, but will not scale for large ones. Deduper employs a large scale distributed persistent hashing mechanism (known as the Managed State) which allows
+it to identify if a particular tuple is duplicate or unique.
+Each time it identifies a tuple as a unique tuple, it also
+stores it into the Managed state for future lookup.
+
+![](images/deduper/image04.png)
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+Following are the different components of the Deduper
+
+1.  **Dedup Operator** - This is responsible for the overall
+    functionality of the operator. This in turn makes use of other
+    components to establish the end goal of deciding whether a tuple is
+    a duplicate of some earlier tuple, or is a unique tuple.
+2.  **Managed State** - Since, all of the data cannot be stored in
```

Comment:
May be worth mentioning that this will also maintain in-memory cache for repeated access. Also do we say something how state that is no longer needed is discarded?